pageserver/src/tenant.rs

```diff
@@ -1228,6 +1228,9 @@ impl Tenant {
 
         if self.get_timeline(new_timeline_id, false).is_ok() {
             debug!("timeline {new_timeline_id} already exists");
+            // FIXME: in this case we should probably still await that the index_part.json upload
+            // complete to be retryable, assuming timeout happened with `branch_timeline` waiting
+            // for upload part; cannot see any good candidates
             return Ok(None);
```

Comment:
Highlighting fixme. I don't think it'd be bad to await on the `Conflict` returning case as long as the Conflict response will not be retried.

Reply:
Added in [5a36e52](https://github.com/neondatabase/neon/pull/4204/commits/5a36e52d7c26abef43f5a1a2bc0c62da1a8dc226).